pkg/store/proxy.go

```diff
@@ -344,11 +398,55 @@ func startStreamSeriesSet(
 
 	wg.Add(1)
 	go func() {
+		isFirstRecv := true
 		defer wg.Done()
 		defer close(s.recvCh)
-
 		for {
-			r, err := s.stream.Recv()
+			var cancel context.CancelFunc
+			if s.responseTimeout != 0 {
+				ctx, cancel = context.WithTimeout(ctx, s.responseTimeout)
+				defer cancel()
+			}
+			rCh := make(chan *recvResponse, 1)
```

Comment:
Instead of allocating this every time we can reuse this channel

Reply:
We need to wait last recv on timeout for close channel for send unused data frame, therefore we can't reuse firs channel